source/dotnet/wholesale-api/Infrastructure/EventPublishers/IntegrationEventFactory.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.Core.JsonSerialization;
+using Energinet.DataHub.Wholesale.Application;
+using Energinet.DataHub.Wholesale.Application.Processes.Model;
+using Energinet.DataHub.Wholesale.Contracts.Events;
+using Energinet.DataHub.Wholesale.Domain.ProcessStepResultAggregate;
+using Energinet.DataHub.Wholesale.Infrastructure.Integration;
+using Google.Protobuf;
+using NodaTime;
+
+namespace Energinet.DataHub.Wholesale.Infrastructure.EventPublishers
+{
+    public class IntegrationEventFactory : IIntegrationEventFactory
+    {
+        private readonly IClock _systemDateTimeProvider;
+        private readonly ICalculationResultReadyIntegrationEventFactory _calculationResultReadyIntegrationEventFactory;
+        private readonly IJsonSerializer _jsonSerializer;
+        private readonly IIntegrationEventTypeMapper _integrationEventTypeMapper;
+
+        public IntegrationEventFactory(
+            IClock systemDateTimeProvider,
+            ICalculationResultReadyIntegrationEventFactory calculationResultReadyIntegrationEventFactory,
+            IJsonSerializer jsonSerializer,
+            IIntegrationEventTypeMapper integrationEventTypeMapper)
+        {
+            _systemDateTimeProvider = systemDateTimeProvider;
+            _calculationResultReadyIntegrationEventFactory = calculationResultReadyIntegrationEventFactory;
+            _jsonSerializer = jsonSerializer;
+            _integrationEventTypeMapper = integrationEventTypeMapper;
+        }
+
+        public IntegrationEventDto CreateIntegrationEventForCalculationResultForEnergySupplier(ProcessStepResult processStepResult, ProcessCompletedEventDto processCompletedEventDto, string energySupplierGln)
```

Comment:
To supplement my interface comment. I'd rather like/expect to see a method something like (together with an interface rename to `ICalculationResultCompletedFactory`):
```suggestion
        public CalculationResultCompletedDto CreateForEnergySupplier(ProcessStepResult processStepResult, ProcessCompletedEventDto processCompletedEventDto, string energySupplierGln)
```

Reply:
Renamed.